internal/services/v1/watch.go

```diff
@@ -107,6 +103,24 @@ func (ws *watchServer) Watch(req *v1.WatchRequest, stream v1.WatchService_WatchS
 						return status.Errorf(codes.Canceled, "watch canceled by user: %s", err)
 					}
 				}
+				if len(update.ChangedDefinitions) > 0 || len(update.DeletedCaveats) > 0 || len(update.DeletedNamespaces) > 0 {
+					if err := stream.Send(&v1.WatchResponse{
+						SchemaUpdated:               true,
+						ChangesThrough:              zedtoken.MustNewFromRevision(update.Revision),
+						OptionalTransactionMetadata: update.Metadata,
+					}); err != nil {
+						return status.Errorf(codes.Canceled, "watch canceled by user: %s", err)
+					}
+				}
+				if update.IsCheckpoint {
+					if err := stream.Send(&v1.WatchResponse{
+						IsCheckpoint:                update.IsCheckpoint,
+						ChangesThrough:              zedtoken.MustNewFromRevision(update.Revision),
+						OptionalTransactionMetadata: update.Metadata,
+					}); err != nil {
+						return status.Errorf(codes.Canceled, "watch canceled by user: %s", err)
```

Comment:
could construct the `WatchResponse` above and then only call `send` once, so that we don't have to copy this error message around?

Reply:
What you are suggesting is:

```
WatchResponse{schema_change: ..., is_checkpoint: true}
```

My understanding from @josephschorr is that we want separate messages:

```
WatchResponse{schema_change: ... }
WatchResponse{is_checkpoint: true}
```

Reply:
Yeah, you need a distinct send